recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,399 @@
+        tools.replace_in_file(os.path.join('VisualMagick', 'lqr', 'Config.txt'),
+                              '#define MAGICKCORE_LQR_DELEGATE', '')
+        # FIXME: package LibRaw
+        tools.replace_in_file(os.path.join('VisualMagick', 'libraw', 'Config.txt'),
+                              '#define MAGICKCORE_RAW_R_DELEGATE', '')
+
+        # FIXME: package FLIF (FLIF: Free Lossless Image Format)
+        tools.replace_in_file(os.path.join('VisualMagick', 'flif', 'Config.txt'),
+                              '#define MAGICKCORE_FLIF_DELEGATE', '')
+
+        # FIXME: package libheif (High Efficiency Image File Format)
+        tools.replace_in_file(os.path.join('VisualMagick', 'libheif', 'Config.txt'),
+                              '#define MAGICKCORE_HEIC_DELEGATE', '')
+
+        # FIXME: package librsvg
+        tools.replace_in_file(os.path.join('VisualMagick', 'librsvg', 'Config.txt'),
+                              '#define MAGICKCORE_RSVG_DELEGATE', '')
+
+        if not self.options.shared:
+            for module in self._modules:
+                tools.replace_in_file(os.path.join('VisualMagick', module, 'Config.txt'), '[DLL]', '[STATIC]')
+            tools.replace_in_file(os.path.join('VisualMagick', 'coders', 'Config.txt'), '[DLLMODULE]',
+                                  '[STATIC]\n[DEFINES]\n_MAGICKLIB_')
+
+        if self.settings.arch == 'x86_64':
+            project = os.path.join('VisualMagick', 'configure', 'configure.vcxproj')
+            tools.replace_in_file(project, 'Win32', 'x64')
+            tools.replace_in_file(project, '/MACHINE:I386', '/MACHINE:x64')
+
+        with tools.chdir(os.path.join('VisualMagick', 'configure')):
+
+            toolset = self.settings.get_safe("compiler.toolset")
+            if not toolset:
+                toolset = {'12': 'v120',
+                           '14': 'v140',
+                           '15': 'v141'}.get(str(self.settings.compiler.version))
+            tools.replace_in_file('configure.vcxproj',
+                                  '<PlatformToolset>v120</PlatformToolset>',
+                                  '<PlatformToolset>%s</PlatformToolset>' % toolset)
+
+            msbuild = MSBuild(self)
+            # fatal error C1189: #error:  Please use the /MD switch for _AFXDLL builds
+            msbuild.build_env.flags = ["/MD"]
+            msbuild.build(project_file='configure.vcxproj',
+                          platforms={'x86': 'Win32'}, force_vcvars=True)
+
+            # https://github.com/ImageMagick/ImageMagick-Windows/blob/master/AppVeyor/Build.ps1
+            command = ['configure.exe', '/noWizard']
+            msvc_version = {9: '/VS2002',
+                            10: '/VS2010',
+                            11: '/VS2012',
+                            12: '/VS2013',
+                            14: '/VS2015',
+                            15: '/VS2017'}.get(int(str(self.settings.compiler.version)))
```

Comment:
What if `self.settings.compiler.version` is 16/17? ==> Visual Studio 2019 / 2022

Reply:
I added these, but it seems like this should be something handled at a higher level.

Reply:
These `/VSYYYY` arguments are imagemagick specific.